app/dao/relation.go

```diff
@@ -0,0 +1,110 @@
+package dao
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"yatter-backend-go/app/domain/object"
+	"yatter-backend-go/app/domain/repository"
+
+	"github.com/jmoiron/sqlx"
+)
+
+type (
+	// Implementation for repository.Relation
+	relation struct {
+		db *sqlx.DB
+	}
+)
+
+// Create relation repository
+func NewRelation(db *sqlx.DB) repository.Relation {
+	return &relation{db: db}
+}
+
+func (r *relation) AddRelation(ctx context.Context, relation object.Relation) error {
+	_, err := r.db.ExecContext(ctx, "INSERT INTO relation (follower_id, followee_id) VALUES (?, ?) ON DUPLICATE KEY UPDATE follower_id = follower_id", relation.FollowerId, relation.FolloweeId)
+
+	if err != nil {
+		return fmt.Errorf("%w", err)
+	}
+	return nil
+}
+
+func (r *relation) FindRelationById(ctx context.Context, follower_id int64, followee_id int64) (*object.Follow, error) {
+	follow := new(object.Follow)
+	relationship := new(object.Relation)
+	err := r.db.QueryRowxContext(ctx, "select * from relation where follower_id = ? and followee_id = ?", follower_id, followee_id).StructScan(relationship)
+	if err == nil {
+		follow.Following = true
+	} else if errors.Is(err, sql.ErrNoRows) {
+		follow.Following = false
+	} else {
+		return nil, fmt.Errorf("%w", err)
+	}
```

Comment:
この実装でも問題ありませんが、ハンドリングが複雑化してしまうので以下のようにすると良いかもしれません

```suggestion
        var exists bool
	if err := r.db.QueryRowxContext(ctx, "select exists ( select * from relation where follower_id = ? and followee_id = ?", follower_id, followee_id).Scan(&exists); err != nil {
		return nil, fmt.Errorf("%w", err)
	} 
	
	follow.Following = exists
```